lib/sarif/pattern_search_sarif.rb

```diff
@@ -0,0 +1,97 @@
+require 'salus/bugsnag'
+
+module Sarif
+  class PatternSearchSarif < BaseSarif
+    include Salus::SalusBugsnag
+
+    NOT_FOUND = "Required Pattern Not Found".freeze
+    PATTERN_SEARCH_URI = "https://github.com/coinbase/salus/blob/master/docs/scanners/"\
+    "pattern_search.md".freeze
+
+    def initialize(scan_report)
+      super(scan_report)
+      @uri = PATTERN_SEARCH_URI
+      @logs = parse_scan_report!
+    end
+
+    def parse_scan_report!
+      scan_hash = @scan_report.to_h
+      hits = scan_hash.dig(:info, :hits)
+      misses = scan_hash.dig(:info, :misses)
+      hits.concat(misses)
+    end
+
+    def build_result(parsed_issue)
+      result = super
+      uri = result[:locations][0][:physicalLocation][:artifactLocation][:uri]
+      result[:locations] = [] if uri.nil?
+      result
+    end
+
+    def build_rule(parsed_issue)
+      rule = super
+      return if rule.nil?
+
+      rule[:fullDescription][:text] = NOT_FOUND if rule[:id] == NOT_FOUND
+      rule
+    end
+
+    def message(hit, miss)
+      pattern = hit[:regex]
+      msg = hit[:msg]
+
```

Comment:
In the conditional below, you do a bunch of `nil?` checks, like `if !pattern.nil?` and `if !msg.nil?`. These things could never be nil, they could only be strings or non-empty strings.
Could you update these to check if they are empty strings?  Like `if pattern != ""`?